doc/paikannuspohjan_kontekstit.md

```diff
@@ -53,25 +66,60 @@ Myös käsitteiden keskinäiset viittaukset tehdään aina tällä virallisella
 julkaisussa. Tämä varmistaa että virallinen ID säilyy muita viittaajia varten kaikissa ketjuissa.
 
+
+### Käsitteiden viitteet ja kontekstin esitys tietokannassa
+Alla oleva taulukko kuvaa eri kontekstien esitystavat tietokannan sarakkeissa, sekä virallisen ID:n määräytymisen.
+
+| Konteksti                 | Virallinen ID                                | draft | design_id | official_row_id | design_row_id |
+|---------------------------|----------------------------------------------|-------|-----------|-----------------|---------------|
+| Virallinen paikannuspohja | id                                           | false | null      | null            | null          |
+| Luonnos paikannuspohja    | coalesce(official_row_id, id)                | true  | null      | X / null        | X / null      |
+| Julkaistu suunnitelma     | coalesce(official_row_id, id)                | false | X         | X / null        | null          |
+| Luonnos suunnitelma       | coalesce(official_row_id, design_row_id, id) | false | X         | X / null        | X / null      |
+
+#### Virallinen paikannuspohja
+- Virallinen käsite ei voi koskaan viitata muihin konteksteihin, eikä siihen voi liittyä suunnitelmaa
+ 
+#### Luonnos paikannuspohja
+- Jos official_row_id on määritelty, kyseessä on luonnosmuutos, muutoin kyseessä on uusi luonnos
+- Jos design_row_id on määritelty, draft on luotu toteuttamalla se suunnitelmasta
+  - Julkaisun yhteydessä sekä suunnitelmarivi että luonnosrivi poistuu, sillä kyse on suunnitelman valmistumisesta
+
+#### Julkaistu suunnitelma
+- Jos official_row_id on määritelty, kyseessä on muutossuunnitelma. Muutoin kyseessä on uuden olion suunnitelma.
+ 
+#### Luonnos suunnitelma
+- Jos design_row_id on määritelty, kyseessä on luonnosmuutos olemassaolevaan suunnitelmaan, muutoin kyseessä on uusi luonnos
+- Jos official_row_id on määritelty, kyseessä on suunniteltu muutos viralliseen paikannuspohjaan, muutoin kyseessä on uuden käsitteen suunnitelma
+- Huom. kaikki yhdistelmät official_row_id:n ja design_row_id:n kanssa ovat mahdollisia:
+  - Uusi luonnossuunnitelma uudelle raiteelle: ei kumpaakaan määritelty
+  - Uusi luonnossuunnitelma olemassaolevalle raiteelle: vain official_row_id määritelty
+  - Muokattu luonnossuunnitelma uudelle raiteelle: vain design_row_id määritelty
+  - Muokattu luonnossuunitelma olemassaolevalle raiteelle: molemmat määritelty
+
 ### Kaavioiden merkinnät
 
 ```mermaid
 graph TD
     subgraph legendColor [Värikoodit]
-        legendDraft(Luonnos olio)
-        legendOfficial(Virallinen olio)
-        legendPlan(Suunniteltu olio)
-        legendDraft ~~~ legendOfficial ~~~ legendPlan
+        legendDraft(Luonnos paikannuspohja)
+        legendOfficial(Virallinen paikannuspohja)
+legendDraftPlan(Luonnos suunnitelma)
+        legendPlan(Julkaistu suunnitelma)
+        legendDraft ~~~ legendOfficial ~~~ legendDraftPlan ~~~ legendPlan
```

Comment:
Näissä kanssa "Luonnospaikannuspohja" ja "Luonnossuunnitelma"

Reply:
En nyt pysty pistämään kommentteja noihin "Viite olioiden välillä", "Olio siirtyy" jne., mutta mielestäni niihin voisi kommentoida vielä eksplisiittisesti tyyliin "katkoviiva", "pisteviiva" jne. minkälaista viivaa ne kuvaavat. Itse en nopealla silmäyksellä meinannut ensin huomata että kyseessähän on oikeasti erilaiset viivat ja etsin eroja muualta.

Reply:
Lisäkommenttina jotenkin tuo "Olio siirtyy" -käsite hämää myös minua. Jotenkin olettaisin että kun puhutaan siirtymisestä niin olio oikeasti vaihtaisi jotekin paikkaa mutta pysyisi muuten muuttumattomana, mutta näin selkeästi ei näyttäisi olevan, vaan sikäli kun luen kaaviota oikein, niin sitä näytettäisiin käytettävän indikoimaan sitä, että jokin asset on sama asset kuin edellisessä askeleessa.

Reply:
Hyviä pointteja. Lisäsin viivoihin selitteet ja vaihdoin termit muotoon "olion kantarivi säilyy" vs "olion data kopioidaan toiselle riville". Ehkä jokseenkin selkeämpi mitä tuolla haetaan. Tosiaan ne eivät säily välttämättä muuttamattomina (kontekstivaihdon yhteydessä on mahdollista että osa datasta muuttuu) mutta pohjalle on aina otettu yhden kontekstin versio ja sen tiedot tallennetaan jollekin käsitteen esiintymistä - joissain siirtymissä ylikirjoittaen toisen rivin, joissain päivittäen omaa riviä, joissain luoden rinnalle uuden kopion.